### ISSUE TITLE
Gene Ontology search doesn't work

### ISSUE THREAD

**Issue by zsloan** [genenetwork/genenetwork2#92]
I think that the GO search isn't working because the db_GeneOntology database doesn't exist on the penguin mysql server. I think that I might also need to make other changes to the way it creates the GO query, due to it querying within 2 different databases.

**Comment by lyan6** [comment]
I have transferred the database db_GeneOntology from Lily to Penguin.

**Comment by zsloan** [comment]
After Lei added the db_GeneOntology database GO searches are working, and I just committed a fix that allows GO searches to be used in combined searches with other terms. The only problem is that the GO search returns far fewer results than its GN1 equivalent. Lei said something about the logic for this search having been changed in GN1, so I think that it's currently using whatever the old logic was in GN2.

**Comment by robwwilliams** [comment]
Dear Zach,

Do we have the Genotation link implemented in GN2?

On Fri, Jul 10, 2015 at 4:05 PM, zsloan <notifications@github.com> wrote:

> After Lei added the db_GeneOntology database GO searches are working, and
> I just committed a fix that allows GO searches to be used in combined
> searches with other terms. The only problem is that the GO search returns
> far fewer results than its GN1 equivalent. Lei said something about the
> logic for this search having been changed in GN1, so I think that it's
> currently using whatever the old logic was in GN2.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/genenetwork/genenetwork2/issues/92#issuecomment-120526425>
> .
>



-- 
Rob

Robert W. Williams, Ph.D.
UT-ORNL Governor's Chair in Computational Genomics
Chair, Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
Room 501
855 Monroe Avenue, Memphis TN 38163 USA

Office 901 448-7018  CELL 901
Office: 501 Wittenborg Building
EMAIL: rwilliams@uthsc.edu
Alternative email: labwilliams@gmail.com
SKYPE: robwwilliams

**Comment by zsloan** [comment]
Yes, it's the first resource link on the trait page for any trait that has
a gene symbol associated with it ( for example
http://gn2.genenetwork.org/show_trait?trait_id=236263_at&dataset=GSE5281_F_RMA_N_0709
).

**Comment by zsloan** [comment]
This old issue was apparently never fixed - GO search returns fewer results in GN2 than in GN1. I need to look at the GN1 logic and see what is different.

**Comment by chfi** [comment]
@zsloan is this resolved?

**Comment by zsloan** [comment]
Sort of. Basic GO searches were added into GN2 after this issue was posted
in a way that produces results very similar (but not identical) to GN1, but
allowing combined GO searches (so searching for stuff with a specific GO Id
*and* a Max LRS > X, etc) currently won't work correctly.

It's a bit tricky because the GN1 search code is difficult to parse and GO
searches function differently from most other searches (because they have
an intermediary step where they convert the GO ID into a list of gene
symbols, which is basically what I did to get it working, though I don't
know why this results in slightly different results from GN1).
